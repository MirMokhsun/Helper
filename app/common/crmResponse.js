import { localization } from "./localization";

export const crmResponseMassage = (statusCode, errorFromServer, stor) => {
  const local = stor.getState().currentLanguage.language;
  const currentLang = localization[local];
  let statusResponse = '';
  if (statusCode === 401) {
    // Email or Password Incorrect
    statusResponse = currentLang.EmailOrPassWrong;
  }
  else if (statusCode === 500 && errorFromServer === 3) {
    // Invalid parametrs
    statusResponse = currentLang.InvalidParameters;
  }
  else if (statusCode === 500 && errorFromServer === 6) {
    // Email already exist
    statusResponse = currentLang.EmailIsExist;
  }
  else if (statusCode === 500 && errorFromServer === 10) {
    // Email not exist
    statusResponse = currentLang.EmailWrong;
  }
  else {
    // All another errors from server
    statusResponse = currentLang.AnotherProblem;
  }

  return statusResponse;
}