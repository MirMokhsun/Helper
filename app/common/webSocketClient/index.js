import WebSocketClient from './WebSocketClient';

let instance = null;
export class WebSocketSingleton {
    constructor(url) {
        if (!instance && url) {
            instance = this;
            this.webSocketClient = new WebSocketClient({ wsUrl: url });
        }
        return instance;
    }
}

// export const webSocketClient = new WebSocketSingleton().webSocketClient;
