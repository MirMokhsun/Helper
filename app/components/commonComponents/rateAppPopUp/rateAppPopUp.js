/* eslint-disable no-plusplus */
import React from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { Modal } from 'react-native';
import { RatePopup } from './ratePopUp/ratePopUp';
import { TopComponent } from './ratePopUp/topComponent/topComponent';
import Utils from '../../../api/utils/utils';

export class RateAppPopUp extends React.Component {
    constructor(props) {
        super(props);
        this.whenShowPopUp = [6, 20, 30];
        this.state = {
            isShowPopUp: false,
            openAppCounter: 0,
            newState: null,
            is10SecondsPassed: false,
        }
    };

    closePopUp = () => {
        this.setState({ isShowPopUp: false });
    }

    mountingRateAppPopUp = (string) => {
        const rateAppObject = JSON.parse(string);

        if (!Utils.chackIsObjectEmpty(rateAppObject)) { // если обект не пуст
            if (!rateAppObject.isClientRateApp) { // и клиент не оценил его
                setTimeout(() => {
                    const newState = this.createNewState(rateAppObject);
                    this.setState({ newState, is10SecondsPassed: true });
                }, 10000);
            }
        }
    }

    writeDataRate = async () => {
        try {
            const { newState } = this.state;
            if (newState) {
                this.setState(newState);
                await AsyncStorage.setItem('@rateApp:key', JSON.stringify(newState));
            }
        } catch (error) {

        }
    }

    firstMountingRateAppPopUp = async () => {
        try {
            const rateAppData = {
                isShowPopUp: false,
                openAppCounter: 1,
                showPopUpcounter: 0,
                isClientRateApp: false,
            };
            await AsyncStorage.setItem('@rateApp:key', JSON.stringify(rateAppData));
        } catch (error) {

        }
    }

    componentDidUpdate() {
        const { is10SecondsPassed, newState, openAppCounter } = this.state;
        if (openAppCounter === 0 && is10SecondsPassed && newState) {
            this.writeDataRate();
        }
    }

    componentDidMount = async () => {
        try {
            await AsyncStorage.getItem('@rateApp:key').then((obj) => {
                if (obj) {
                    this.mountingRateAppPopUp(obj);
                } else {
                    this.firstMountingRateAppPopUp();
                }
            });
        } catch (error) {
            console.warn(error);
        }
    }

    createNewState = (rateAppObject) => {
        const _openAppCounter = ++rateAppObject.openAppCounter;
        const { showPopUpcounter } = rateAppObject;
        const _isShowPopUp = this.checkIfNeedToShowPopUp(_openAppCounter, showPopUpcounter);
        const _showPopUpcounter = this.checkIfPopUpShown(_isShowPopUp, showPopUpcounter);
        const _isClientRateApp = false;

        return {
            isShowPopUp: _isShowPopUp,
            openAppCounter: _openAppCounter,
            showPopUpcounter: _showPopUpcounter,
            isClientRateApp: _isClientRateApp,
        }
    }

    checkIfNeedToShowPopUp = (_openAppCounter, showPopUpcounter) => {
        let isShowPopUp = false;
        for (let i = showPopUpcounter; i < this.whenShowPopUp.length; ++i) {
            if (this.whenShowPopUp[i] === _openAppCounter) {
                isShowPopUp = true;
            }
        }
        return isShowPopUp;
    }

    checkIfPopUpShown = (_isShowPopUp, _showPopUpcounter) => {
        let showPopUpcounter = _showPopUpcounter;
        if (_isShowPopUp) {
            ++showPopUpcounter;
        }
        return showPopUpcounter;
    }

    clientRatedApp = async () => {
        try {
            await AsyncStorage.setItem('@rateApp:key', JSON.stringify({
                isShowPopUp: false,
                openAppCounter: 0,
                showPopUpcounter: 0,
                isClientRateApp: true,
            }));
            this.closePopUp();
        } catch (error) {

        }
    }

    render() {
        const { CurrentLanguage } = this.props;
        const { isShowPopUp } = this.state;
        return (
            <Modal
                visible={isShowPopUp}
                animationType="fade"
                transparent
                onRequestClose={this.closePopUp}
            >
                <RatePopup
                    width="90%"
                    topComponent={<TopComponent
                        title={CurrentLanguage["Liked the application"]}
                        sendText={CurrentLanguage['sendBtnText']}
                        placeholderText={CurrentLanguage["Leave a comment"]}
                        closeFunc={this.clientRatedApp}
                        errorText={CurrentLanguage["10 characters minimum"]}
                    />}
                    backgroundFunction={this.closePopUp}
                    textBtn={CurrentLanguage["Remind me later"]}
                    functionBtn={this.closePopUp}
                />
            </Modal>
        );
    }
}