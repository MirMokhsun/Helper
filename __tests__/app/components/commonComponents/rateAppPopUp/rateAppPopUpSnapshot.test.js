import 'react-native';
import React from 'react';
import { RateAppPopUp } from '../../../../../app/components/commonComponents/rateAppPopUp/rateAppPopUp';
import { localization } from '../../../../../app/common/localization';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('RateAppPopUp', () => {
    it('create snapshot rateAppPopUp', () => {
        const renderer = new ShallowRenderer();
        let rateAppPopUp = renderer.render(<RateAppPopUp CurrentLanguage={localization.ru} />);
        expect(rateAppPopUp).toMatchSnapshot();
    });
});
