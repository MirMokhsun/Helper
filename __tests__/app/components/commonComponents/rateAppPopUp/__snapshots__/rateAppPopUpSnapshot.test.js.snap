// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RateAppPopUp create snapshot rateAppPopUp 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <RatePopup
    backgroundFunction={[Function]}
    functionBtn={[Function]}
    textBtn="Напомнить позже"
    topComponent={
      <TopComponent
        closeFunc={[Function]}
        errorText="Минимальное количество 10 символов"
        placeholderText="Оставить комментарий"
        sendText="Отправить"
        title="Понравилось приложение?
Оцените его!"
      />
    }
    width="90%"
  />
</Component>
`;
